File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/Antidote.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;
import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class Antidote extends Equipable {
	{
		name = Locale.get("antidote");
		description = Locale.get("antidote_desc");
		sprite = "item-antidote";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).poisonResist = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).poisonResist = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/Antidote.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class Antidote : Equipable() {
	init {
		name = Locale.get("antidote")
		description = Locale.get("antidote_desc")
		sprite = "item-antidote"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.poisonResist = ((this.owner.poisonResist + 1) % 128).toByte()
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.poisonResist = ((this.owner.poisonResist - 1) % 128).toByte()
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}